### PR TITLE
docs: switch to local search

### PR DESCRIPTION
**Pull request by userquin**
This PR also includes:
- bump VP to beta 2
- bump `vite-plugin-pwa` to 0.16.4
- bump `@vite-pwa/vitepress` to 0.2.0
- included `@vite-pwa/assets-generator` to rebuild pwa icons: re-built all pwa icons
- remove tab styles
- included `code-groups` in guide index for Vitest installation using `npm/yarn/pnpm`
- prefetch avatars and sponsors images in home page
- removed `localeLinks`
- ~~exclude `README.md` and `CONTRIBUTING.md` files in VP config file~~
- configure search properly: we can switch to Algolia search in this PR
- included `screenshots`, `handle_links` and `launch_handler` in pwa manifest
- included pwa icons: transparent 64x64, white 512x512 for any purpose, change transparent 512x512 purpose to any
- included new transparent 64x64 favicon and update head page link

### CODE REVIEW

**Review by sheremet-va — APPROVED** [review]

**Review by dammy001 — APPROVED** [review]

File: docs/.vitepress/config.ts [hunk 1]
```diff
@@ -21,21 +21,22 @@ export default withPwa(defineConfig({
   lang: 'en-US',
   title: vitestName,
   description: vitestDescription,
+  srcExclude: ['README.md', 'CONTRIBUTING.md'],
```
Comment (userquin) [inline docs/.vitepress/config.ts:24]: can be removed, those files are in project root

**Comment by userquin** [comment]
Including 2 new fields in the pwa web manifest...

**Comment by sheremet-va** [comment]
@userquin there are conflicts after renovate merge, I'm sorry. Can you handle it and then we can merge this PR

**Comment by userquin** [comment]
sure

**Pull request merged**